amd64/src/semantic.rs

```diff
@@ -747,7 +755,7 @@ pub fn call(a: Rvalue) -> Result<(Vec<Statement>, JumpSpec)> {
         call ?, new_rip:64;
     });*/
     let stmts = rreil!{
-        call ?, (a);
```

Comment:
The new enum doesn't worry me so much as the variable length `Vec<Rvalue>` and `Vec<Lvalue>`; if it truly is variable length i believe it will force the encoding to not be fixed width anymore, which substantially increases the complexity.

It will also likely make a `Vec<BinaryStatement>` impossible, and instead will require `BinaryStatements`, which is its own vec of bytes with a size header, etc.